presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -45,6 +45,7 @@
     private final AtomicBoolean escapeProcessing = new AtomicBoolean(true);
     private final AtomicBoolean closeOnCompletion = new AtomicBoolean();
     private final AtomicReference<PrestoConnection> connection;
+    private final AtomicReference<Optional<StatementClient>> executingClient = new AtomicReference<>(Optional.empty());
```

Comment:
I'd make this just  `AtomicReference<StatementClient>` and then later do `Optional.ofNullable()` if you really want the helper.